opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/authenticate/embedded/Xs2aPisReportSelectedScaMethod.java

```diff
@@ -51,6 +53,9 @@ protected void doRealExecution(DelegateExecution execution, Xs2aPisContext conte
                 params.getBody()
         );
 
+        byte[] imageData = paymentRepository.findByPsuId(context.getPsuId()).getImageData();
```

Comment:
You should read this from Xs2aPisContext

Reply:
Now it is written to the context